### ISSUE TITLE
Will you update the installer to new versions of cjdns?

### ISSUE THREAD

**Issue by lessneek** [interfect/cjdns-installer#4]

**Comment by interfect** [comment]
Yes I will! See https://github.com/interfect/cjdns-installer/releases/tag/v0.4-proto16 which I have just released.

Please open another issue if it gives you any trouble.

**Comment by lessneek** [comment]
Now it doesn't work on Windows10. The service starts and works and even connects to other nodes via UDP, but TAP adapter is unplugged, hence there is no IPv6 of cjdns. I tried TAP 9.9.2 and 9.21.1.

**Comment by interfect** [comment]
I've never tested it on Windows 10. Did the previous release work on
Windows 10?

Do you happen to have Kaspersky Antivirus? It is noted to cause trouble in <
https://community.openvpn.net/openvpn/ticket/592>.

Can other applications (OpenVPN, VirtualBox) use the TAP adapter?

Have you tried renaming your TAP adapter to something without any special
characters in it? There's a known issue with adapter names in e.g. Russian.

**Comment by lessneek** [comment]
Yes it worked on win10. I have no any antivirus nor firewall except integrated windows ones. I use English windows and all names are default and in English also. I don't think some other app uses this tap, but I use softether VPN which leverage it's own tap device.

## 

Alexander Logger

**Comment by interfect** [comment]
You might want to go back to the old cjdns build.

Can you get the logs (by running cjdroute --nobg < cjdroute.conf in
the terminal) and post them? It may be complaining about something
obvious.

**Comment by lessneek** [comment]
How can I diagnose this problem? Or better use a linux? =)

**Comment by lessneek** [comment]
It was a little tricky to run with a cmd, I converted 'cjdroute.conf' file encoding to UTF-8 to get it started.
So here is the log:

```
C:\Program Files (x86)\cjdns>cjdroute.exe --nobg < cjdroute.conf
1441613876 INFO cjdroute2.c:585 Cjdns i386 win32
1441613876 INFO cjdroute2.c:589 Checking for running instance...
1441613876 DEBUG UDPAddrIface.c:294 Bound to address [0.0.0.0:56262]
1441613876 DEBUG AdminClient.c:333 Connecting to [127.0.0.1:11234]
1441613876 DEBUG Pipe.c:134 Buffering a message
1441613876 DEBUG cjdroute2.c:634 Sent [140] bytes to core
1441613876 INFO RandomSeed.c:42 Attempting to seed random number generator
1441613876 INFO RandomSeed.c:50 Trying random seed [RtlGenRandom() (Windows)] Success
1441613876 INFO RandomSeed.c:64 Seeding random number generator succeeded with [1] sources
1441613876 INFO LibuvEntropyProvider.c:59 Taking clock samples every [1000]ms for random generator
11441613876 DEBUG 441613876 DEBUG CPore.cipe.c::256 231 GPetting pre-configuration from client
ipe [\\.\pipe\cjdns_pipe_client-core-01cmcxzg4pjuh2hfj3f2cql0p7gvym] established connection
11441613876 DEBUG 441613876 DEBUG PPipe.cipe.c::231 253 PSipe [\\.\pipe\cjdns_pipe_client-core-01cmcxzg4pjuh2hfj3f2cql0p7gvym] established connection
ending buffered message
1441613876 DEBUG Core.c:259 Finished getting pre-configuration from client
1441613876 DEBUG UDPAddrIface.c:255 Binding to address [127.0.0.1:11234]
1441613876 DEBUG UDPAddrIface.c:294 Bound to address [127.0.0.1:11234]
1441613876 DEBUG UDPAddrIface.c:294 Bound to address [0.0.0.0:56263]
1441613876 DEBUG AdminClient.c:333 Connecting to [127.0.0.1:11234]
1441613876 INFO Configurator.c:135 Checking authorized password 0.
1441613876 INFO Configurator.c:157 Adding authorized password #[0] for user [password [0]].
1441613876 WARN Configurator.c:111 Got error [Failed to configure tunnel [WinFail.c:3059 ( RegQueryValueEx(connKey, name_string, NULL, &name_type, (uint8_t*)name_data, &len) ) [ERROR_FILE_NOT_FOUND]]] calling [Core_initTunnel], ignoring.
1441613876 DEBUG Configurator.c:486 Security_chroot(/var/run/)
1441613876 DEBUG Configurator.c:529 Security_noforks()
1441613876 DEBUG Configurator.c:546 Security_seccomp()
1441613876 DEBUG Configurator.c:551 Security_setupComplete()
1441613876 DEBUG Configurator.c:635 Cjdns started in the background
1441613876 DEBUG cjdroute2.c:675 Keeping cjdns client alive because --nobg was specified on the command line
```

**Comment by interfect** [comment]
**TL;DR: [Download this](https://mega.nz/#!rFsg1aJR!tx7kaI1tuCTOzLEnPWnwZ4dt0pbpdcj9vrhgFpoPv0M) and report back.**

To make your logs longer and more informative, ypou can set `"logTo":"stdout"` in the `logging` section of the config file.

OK, from the error message:

```
1441613876 WARN Configurator.c:111 Got error [Failed to configure tunnel [WinFail.c:3059 ( RegQueryValueEx(connKey, name_string, NULL, &name_type, (uint8_t*)name_data, &len) ) [ERROR_FILE_NOT_FOUND]]] calling [Core_initTunnel], ignoring.
```

And from looking at the context of the [relevant code in cjdns here](https://github.com/cjdelisle/cjdns/blob/a05ade40dc31caebaf3aa770aac3ab2ecb02d867/interface/tuntap/windows/TAPDevice.c#L218), where the first error is happening, it looks like cjdns is going after a registry key that doesn't exist.

It's going to the registry key `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Network\{4D36E972-E325-11CE-BFC1-08002BE10318}`, and under that looking in every key that's the right length to be a GUID. Within each of those it's looking for the `Connection` key, and in there the `Name`value. My guess is it's looking for a network connection name that corresponds to the name specified as the TUN adapter in your config file.

The problem is that your registry has a key there that doesn't _have_ a `Name` value. I have no idea _why_ that's the case; maybe Windows 10 has done away with the "Name" value altogether, storing that information elsewhere, or maybe there's just some device with an invalid entry.

It might be useful if you could send or post a dump of the relevant registry key; open up regedit, go to `HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Network\{4D36E972-E325-11CE-BFC1-08002BE10318}`, right-click on it, and export it as a .reg file, open _that_ in Notepad, and copy-paste it in here.

Assuming you don't need to pick a _specific_ TAP device, there's no real reason it needs to read the Name key. 

I've added some code in a [branch here](https://github.com/interfect/cjdns/tree/nameless-interface-tolerance) that tries to proceed if there's no name defined for an interface.

Could you please download [this build](https://mega.nz/#!rFsg1aJR!tx7kaI1tuCTOzLEnPWnwZ4dt0pbpdcj9vrhgFpoPv0M) and see if it manages to find a working TAP interface? There could be some other breaking changes besides the missing names.

**Comment by lessneek** [comment]
Ok. Thanks. With the new exe file it works well.

**Comment by cschmittiey** [comment]
New exe works exactly like it should, thanks interfect! Been hoping this could be made to work for a long time!

**Comment by interfect** [comment]
Excellent. I'll have to rebase it and PR it against mainline, so this can
be fixed for everyone. Also release a new installer...
